File: 프로그래머스/lv2/42586. 기능개발/기능개발.kt
import java.util.*

class Solution {
    fun solution(progresses: IntArray, speeds: IntArray): IntArray {
        var answer:IntArray
        var queue:Queue<Int> = LinkedList<Int>()
        
        /* 개발하는데 필요한 일수 구하기 */
        for(i in 0..progresses.size-1){
            var leftDay = 0

            leftDay = (100-progresses[i])/speeds[i]
            if((100-progresses[i])%speeds[i] != 0){
                leftDay++
            }
            queue.add(leftDay)
        }

        /* Queue의 앞 날짜보다 작은 일수 세기 */
        var result = mutableListOf<Int>()

        while(!queue.isEmpty()) {
            var count = 1
            var first = queue.poll()

            while (!queue.isEmpty() && queue.peek() <= first) {
                queue.remove()
                count++

            }
            result.add(count)
        }
        answer = result.toIntArray()

        return answer
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    const answer = [];
    const days = progresses.map((p, i) => Math.ceil((100 - p) / speeds[i]));

    let current = days[0];
    let count = 1;

    for (let i = 1; i < days.length; i++) {
        if (days[i] <= current) {
            count++;
        } else {
            answer.push(count);
            current = days[i];
            count = 1;
        }
    }
    // 마지막날꺼 배포 
    answer.push(count);
    
    return answer;
}
